README.md

```diff
@@ -1,55 +1,385 @@
-# `Saturn` — parallelism-safe data structures for Multicore OCaml
+[API reference](https://ocaml-multicore.github.io/saturn/)
 
----
+<!--
+```ocaml
+# #thread
+```
+-->
+
+# Saturn — parallelism-safe data structures for Multicore OCaml
 
 This repository is a collection of parallelism-safe data structures for OCaml 5.
 They are contained in two packages:
 
-- `Saturn` that includes every data structures and should be used by default if
-  you just want parallelism-safe data structures..
-- `Saturn_lockfree` that includes only lock-free data structures.
+- Saturn that includes all data structures and should be used by default if you
+  just want parallelism-safe data structures;
+- Saturn_lockfree that includes only lock-free data structures.
+
+It aims to provide an industria-strenght, well-tested (and possibly
+model-checked and verified in the future), well documented and maintained
+parallelism-safe data structure library. We want to make it easier for multicore
+OCaml users to find the right data structures for their uses.
+
+**Saturn** is published on [opam](https://opam.ocaml.org/packages/saturn/) and
+is distributed under the
+[ISC license](https://github.com/ocaml-multicore/saturn/blob/main/LICENSE.md).
+
+[![OCaml-CI Build Status](https://img.shields.io/endpoint?url=https%3A%2F%2Fci.ocamllabs.io%2Fbadge%2Focaml-multicore%2Fsaturn%2Fmain&logo=ocaml&style=flat-square)](https://ci.ocamllabs.io/github/ocaml-multicore/saturn)
+[![GitHub release (latest by date)](https://img.shields.io/github/v/release/ocaml-multicore/saturn?style=flat-square&color=09aa89)](https://github.com/ocaml-multicore/saturn/releases/latest)
+[![docs](https://img.shields.io/badge/doc-online-blue.svg?style=flat-square)](https://ocaml-multicore.github.io/saturn/)
+
+# Contents
+
+- [Installation](#installation)
+- [Introduction](#introduction)
+  - [Provided data structures](#provided-data-structures)
+  - [Motivation](#motivation)
+- [Usage](#usage)
+  - [Data structures with domain roles](#data-structures-with-domain-roles)
+  - [About composability](#about-composability)
+- [Properties and testing](#properties-and-testing)
+  - [Progress properties](#progress-properties)
+  - [Tests](#tests)
+- [Benchmarks](#benchmarks)
+
+# Installation
+
+## Getting OCaml 5.0
+
+You'll need OCaml 5.0.0 or later. Note that `saturn` also works with OCaml 4.14
```

Comment:
Is this `saturn` a command or should it be Saturn, the name of the data structures?